src/yunohost/app.py

```diff
@@ -341,6 +341,109 @@ def app_map(app=None, raw=False, user=None):
     return result
 
 
+def app_changeurl(auth, app, domain, path):
+    """
+    Modify the URL at which an application is installed.
+
+    Keyword argument:
+        app -- Taget app instance name
+        domain -- New app domain on which the application will be moved
+        path -- New path at which the application will be move
+
+    """
+    from yunohost.hook import hook_exec
+
+    installed = _is_installed(app)
+    if not installed:
+        raise MoulinetteError(errno.ENOPKG,
+                              m18n.n('app_not_installed', app=app))
+
+    old_domain = app_setting(app, "domain")
+    old_path = app_setting(app, "path")
+
+    # Normalize path and domain format
+    domain = domain.strip().lower()
+    old_path_trimed = '/' + old_path.strip("/").strip() + '/'
+    path = '/' + path.strip("/").strip() + '/'
+
+    if (domain, path) == (old_domain, old_path_trimed):
+        raise MoulinetteError(errno.EINVAL, m18n.n("app_change_url_identical_domains", domain=domain, path=path))
+
+    # WARNING / FIXME : checkurl will modify the settings
+    # (this is a non intuitive behavior that should be changed)
+    # (or checkurl renamed in reserve_url)
+    app_checkurl(auth, '%s%s' % (domain, path), app)
+
+    if not os.path.exists(os.path.join(APPS_SETTING_PATH, app, "scripts", "change_url")):
```

Comment:
Shouldn't this check happen at the beginning of the function ? Here we already called app_checkurl and compared the old and new domains.